uk_tax_23-24.py

```diff
@@ -239,16 +238,16 @@ def plot_tax_savings_3d(salary_top_range=salary_top_range, max_voluntary_contrib
     ax1 = fig.add_subplot(121, projection='3d')
     ax2 = fig.add_subplot(122, projection='3d')
 
-    incomes = np.arange(0, salary_top_range + 1000, 1000)
-    X, Y = np.meshgrid(incomes, np.linspace(0, max(incomes) * max_voluntary_contrib_percentage, len(incomes)))
+    incomes = np.arange(500, salary_top_range + 500, 1000)
+    X, Y = np.meshgrid(incomes, np.arange(500, max(incomes) * max_voluntary_contrib_percentage, len(incomes)))
```

Comment:
Step size in this `arange` should be £100 or £500, but definitely not len(incomes). I believe it evaluates to 500 right now, but this is by pure luck (because we have 500 incomes, from £500 to £250,000 every £500).
You can just change `len(incomes)` into a fixed number, let's say 100.

Reply:
One more change and we will be good to merge 👍🏻

Reply:
No worries let me push that